### PR TITLE
#18553 Get host and port from url for native tools if connection is by url

### LINKED ISSUE #18553: Dbeaver does not backup Postgresql if the connection details are in the Url.
### Description

Dbeaver does not backup Postgresql if the connection details are in the Url. 
And that is selected to connect with. 

It will try to get the connection details from the Host, Port and Database variables.

### DBeaver Version

Community Edition 22.3.0

### Operating System

Windows 10

### Database and driver

1. Driver : Postgresql JDBC driver
2. Database: Postgresql 11.15

### Steps to reproduce

1. Try to connect to a postgress database with a URL which is not the default. (A dns url, which is not localhost.)
2. Do not change the default Host, Database and Port variables. 
3. Connection succeeds
4. Try to backup the database with the tools. 
5. Its says it fails with connecting to the localhost, which is not in the Url, but defined in the Host.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**
Dump and Restore should work for PostgreSQL and MySQL when connection was configured by specifying host and port and when connection was configured by url,

### CODE REVIEW

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java [hunk 1]
```diff
@@ -91,4 +100,39 @@ public static String replaceVariables(@NotNull String name) {
...
+                try {
+                    URI uri = URI.create(url);
+                    hostname = uri.getHost();
+                    port = String.valueOf(uri.getPort());
+                } catch (Exception e) {
+                    log.error("Can't parse connection URL", e);
+                }
+            }
+        }
+        cmd.add("--host=" + hostname);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java:133]: What if `hostname` is empty, too?

File: plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java [hunk 2]
```diff
@@ -91,4 +100,39 @@ public static String replaceVariables(@NotNull String name) {
                 return System.getProperty(name);
         }
     }
+    
+    /**
+     * Appends <code>--host</code> and <code>--port</code> parameters from connection settings to the list of command parameters
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java]:
```suggestion
     * Appends {@code --host} and {@code --port} parameters from connection settings to the list of command parameters
```

File: plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java:133]: It was not supposed to be empty. I decided to let this behaviour be untouched.

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/NativeToolUtils.java:133]: Actually, I think it's not possible as it's not a valid URI anymore.

**Comment by Matvey16** [comment]
Verified

**Pull request merged**